virttest/qemu_devices/qcontainer.py

```diff
@@ -427,10 +427,18 @@ def execute_qemu(self, options, timeout=5):
         """
         if self.__execute_qemu_last != options:
             cmd = "%s %s 2>&1" % (self.__qemu_binary, options)
-            self.__execute_qemu_out = str(process.run(cmd, timeout=timeout,
-                                                      ignore_status=True,
-                                                      shell=True,
-                                                      verbose=False).stdout)
+            result = process.run(cmd, timeout=timeout,
+                                 ignore_status=True,
+                                 shell=True,
+                                 verbose=False)
+            if result.exit_status and "machine specified" in result.stdout:
+                # Arm requires machine to be specified, let's try again
+                # with dummy "virt" machine
+                result = process.run("%s -machine virt %s 2>&1"
+                                     % (self.__qemu_binary, options),
+                                     ignore_status=True, shell=True,
+                                     verbose=False)
+            self.__execute_qemu_out = str(result.stdout)
```

Comment:
I don't like the try-again-with-another-config approach, but indeed it will work in this case. You should include a #TODO here with the directions to improve this afterwards. We probably should detect the right machine to use here, but it can be improved later.